src/crypto.c

```diff
@@ -601,9 +627,13 @@ bool message_derive(Scalar out, const Keypair *kp, const ROInput *input, const u
 
     // blake2b hash
     cx_blake2b_t ctx;
-    cx_blake2b_init(&ctx, 256);
-    cx_hash(&ctx.header, 0, derive_msg, derive_len, NULL, 0);
-    cx_hash(&ctx.header, CX_LAST, NULL, 0, out, ctx.ctx.outlen);
+    cx_err_t cx_err;
+    cx_err = cx_blake2b_init_no_throw(&ctx, 256);
+    LEDGER_ASSERT(cx_err == CX_OK, "cx_blake2b_init_no_throw fail");
+    cx_err = cx_hash_no_throw(&ctx.header, 0, derive_msg, derive_len, NULL, 0);
+    LEDGER_ASSERT(cx_err == CX_OK, "cx_hash_no_throw fail");
+    cx_err = cx_hash_no_throw(&ctx.header, CX_LAST, NULL, 0, out, ctx.ctx.outlen);
+    LEDGER_ASSERT(cx_err == CX_OK, "cx_hash_no_throw fail");
```

Comment:
Why `LEDGER_ASSERT`, and not returning `false`, letting the caller to decide what needs to be done?

Reply:
OK 👍